core/src/main/java/jenkins/model/JenkinsLocationConfiguration.java

```diff
@@ -29,6 +29,7 @@
  * @since 1.494
  */
 @Extension @Symbol("location")
```

Comment:
I'd suggest that all Extensions should be implicitly annotated in such way with an opt-out option